core/modules/link/tests/link.attribute.test

```diff
@@ -0,0 +1,494 @@
+    $this->zebra = 0;
+    // Create and login user.
+    $this->web_user = $this->backdropCreateUser(array('administer content types'));
+    $this->backdropLogin($this->web_user);
+  }
+
+  protected function createLink($url, $title, $attributes = array()) {
+    return array(
+      'url' => $url,
+      'title' => $title,
+      'attributes' => $attributes,
+    );
+  }
+
+  protected function assertLinkOnNode($field_name, $link_value, $message = '', $group = 'Other') {
+    $this->zebra++;
+    $zebra_string = ($this->zebra % 2 == 0) ? 'even' : 'odd';
+    $cssFieldLocator = 'field-' . str_replace('_', '-', $field_name);
+    $this->assertPattern('@<div class="field field-type-link ' . $cssFieldLocator . '".*<div class="field-item ' . $zebra_string . '">\s*' . $link_value . '\s*</div>@is',
+                         $message,
+                         $group);
+  }
+
+  /**
+   * A simple test that just creates a new node type, adds a link field to it, creates a new node of that type, and makes sure
+   * that the node is being displayed.
+   */
+  function testBasic() {
+    $content_type_friendly = $this->randomName(20);
+    $content_type_machine = strtolower($this->randomName(10));
+    $title = $this->randomName(20);
+
+    $this->backdropGet('admin/structure/types');
+
+    // Create the content type.
+    $this->clickLink(t('Add content type'));
+
+    $edit = array(
+      'name' => $content_type_friendly,
+      'type' => $content_type_machine,
+    );
+    $this->backdropPost(NULL, $edit, t('Save and add fields'));
+    $this->assertText(t('The content type @name has been added.', array('@name' => $content_type_friendly)));
+
+    // Now add a singleton field.
+    $single_field_name_friendly = $this->randomName(20);
+    $single_field_name_machine = strtolower($this->randomName(10));
+    $single_field_name = 'field_' . $single_field_name_machine;
+    $edit = array(
+      'fields[_add_new_field][label]' => $single_field_name_friendly,
+      'fields[_add_new_field][field_name]' => $single_field_name_machine,
+      'fields[_add_new_field][type]' => 'link_field',
+      'fields[_add_new_field][widget_type]' => 'link_field',
+
+    );
+    $this->backdropPost(NULL, $edit, t('Save'));
+
+    // We'll go with the default settings for this run-through.
+    $this->backdropPost(NULL, array(), t('Save field settings'));
+
+    // Using all the default settings, so press the button.
+    $this->backdropPost(NULL, array(), t('Save settings'));
+    $this->assertText(t('Saved @name configuration.', array('@name' => $single_field_name_friendly)));
+
+    // Somehow clicking "save" isn't enough, and we have to do a
+    // node_types_rebuild().
+    node_types_rebuild();
+    menu_rebuild();
+    $type_exists = db_query('SELECT 1 FROM {node_type} WHERE type = :type', array(':type' => $content_type_machine))->fetchField();
+    $this->assertTrue($type_exists, 'The new content type has been created in the database.');
```

Comment:
@quicksketch looks like this is checking to see if a content type can be created; but it is looking in the `node_type` table.

Is it safe to leave this test out as this is in core now and `node` and `user` modules will be performing these checks anyways?

I've tried to restructure this query against the `node` table but it was not passing tests ... should I keep at it or are these tests moot now for link in core?